subprojects/core-api/src/main/java/org/gradle/api/artifacts/result/ComponentSelectionCause.java

```diff
@@ -65,7 +65,15 @@
     /**
      * This component was selected because of a dependency constraint
      */
-    CONSTRAINT("constraint");
+    CONSTRAINT("constraint"),
+
+    /**
+     * This component was selected because it was requested by a parent with {@link org.gradle.api.artifacts.MutableVersionConstraint#forSubgraph()}.
+     *
+     * @since 5.7
+     */
+    @Incubating
+    BY_PARENT("by parent");
```

Comment:
:o: Should it be `VIA_SUBGRAPH`?

Reply:
When I get this choice, I am _somewhere down in the graph_, where the version constraint was "forced" upon me by something define further up the graph.